### PR TITLE
Adds tests to reproduce applicability test issue

### PR DESCRIPTION
**Pull request by kunli2**
Add tests to reproduce applicability test issue https://github.com/openrewrite/rewrite/issues/2754

### CODE REVIEW

File: rewrite-core/src/test/java/org/openrewrite/text/ChangeTextTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class ChangeTextTest implements RewriteTest {
+    @Test
+    void changeText() {
+        rewriteRun(
+          spec -> spec.recipe(new ChangeText("after")),
+          text("before", "after"),
+          text("", "after")
+        );
+    }
```
Comment (sambsnyd) [inline rewrite-core/src/test/java/org/openrewrite/text/ChangeTextTest.java:31]: I don't think we need this test. `ChangeText` is very simple and already exercised in plenty of tests

File: rewrite-core/src/test/java/org/openrewrite/RecipeLifecycleTest.java [hunk 2]
```diff
@@ -217,6 +218,114 @@ If the applicability test is incorrectly applied to individual recipes in the li
...
+            .activateRecipes("org.openrewrite.RecipesToTransformMultiFiles")),
+          text("1",
+            "1->2->3",
+            spec -> spec.path("file.txt").noTrim())
+        );
+    }
+
+    @Disabled
+    @Test
+    void yamlApplicabilityFalseWithRecipesHaveVisitMethodOverridden() {
```
Comment (sambsnyd) [inline rewrite-core/src/test/java/org/openrewrite/RecipeLifecycleTest.java:277]: If we already have an issue number for this add `@Issue("https://....")`

Comment (kunli2) [reply rewrite-core/src/test/java/org/openrewrite/RecipeLifecycleTest.java:277]: Oh, Yeah, I created an issue ticket after this PR, let me add the @Issue link here.

File: rewrite-core/src/test/java/org/openrewrite/text/ChangeTextTest.java (hunk 1 above)
Comment (kunli2) [reply rewrite-core/src/test/java/org/openrewrite/text/ChangeTextTest.java]: removed.

File: rewrite-core/src/test/java/org/openrewrite/RecipeLifecycleTest.java (hunk 2 above)
Comment (kunli2) [reply rewrite-core/src/test/java/org/openrewrite/RecipeLifecycleTest.java:279]: added.

**Review by sambsnyd — APPROVED** [review]

**Pull request merged**